src/main/java/randoop/main/GenTests.java

```diff
@@ -516,6 +522,75 @@ public boolean handle(String[] args) {
+        testOccurrences = tallyOperationsInSequences(regressionSequences);
+
+        // How many flaky tests an operation occurs in.
+        Map<TypedOperation, Integer> flakyOccurrences;
+
+        // TODO: cxing handle Error Test Sequence tallying.
+        // tallyOperationsInSequences(testOccurrences, explorer.getErrorTestSequences());
+
+        List<ExecutableSequence> flakySequences = new ArrayList<>();
+        for (String flakyTestNum : flakyTests) {
+          int testNum = Integer.parseInt(flakyTestNum.substring(4)); // length of "test"
+          // Tests start at 001
+          ExecutableSequence flakySequence = regressionSequences.get(testNum - 1);
+          flakySequences.add(flakySequence);
+        }
+
+        // Tally occurrences of operations in flaky tests.
+        // Each method is counted exactly once if it appears in a flaky test.
+        flakyOccurrences = tallyOperationsInSequences(flakySequences);
+
+        // if-idf metric
+        // Our heuristic for probability of whether a method is flaky
+        //  is equal to the ratio of the number of flaky tests a method M occurs
+        //  in divided by the number of total tests M occurs in.
+        Map<String, Double> methodFlakyPercentage = new HashMap<>();
+
+        for (TypedOperation to : testOccurrences.keySet()) {
+          double flakyMethodOccurrences =
+              flakyOccurrences.containsKey(to) ? flakyOccurrences.get(to) : 0;
+          methodFlakyPercentage.put(
+              to.toParsableString(), flakyMethodOccurrences / testOccurrences.get(to));
+        }
+
+        // Sort in descending order by value.
+        List<Entry<String, Double>> sortedMethodsByFlakiness =
+            new ArrayList<>(methodFlakyPercentage.entrySet());
+        sortedMethodsByFlakiness.sort(reverseOrder(Entry.comparingByValue()));
+
+        System.out.println();
+        System.out.println("Flaky tests were generated. The following methods,");
+        System.out.println("in decreasing order of likelihood, are the most likely culprits.");
+        System.out.println("Please determine whether the following methods may exhibit");
+        System.out.println("different behavior on separate runs of the same code");
+        System.out.println(
+            "and add those methods to the non-multi-run deterministic method blacklist.");
+        System.out.println("If you are unsure, consider adding one method at a time to the");
+        System.out.println("non-multi-run deterministic method blacklist and re-running Randoop");
+        System.out.println("until no flaky tests are generated.");
+        System.out.println();
+
+        for (int i = 0;
+            i < GenInputsAbstract.num_suspected_flaky_test_to_output
+                && i < sortedMethodsByFlakiness.size();
+            i++) {
+          Entry<String, Double> method = sortedMethodsByFlakiness.get(i);
+          System.out.println(method.getKey() + ":\t " + (method.getValue() * 100.0) + "%");
```

Comment:
Writing this as a percentage seems misleading.  It's not the case that a method with a 100% score is 100% likely to be the cause of the nondeterminism.  And it's not a percentage of anything.  I would state it just as a number for the ranking (without "%" or units).
Also, round it or truncate it (casting to int does truncation) since users are unlikely to care about any digits after the decimal point.